lib/src/license_detection/normalizer.dart

```diff
@@ -0,0 +1,125 @@
+/// Normalizes text according to [SPDX guidelines v2.1][]
+///
+/// [SPDX guidelines v2.1] : https://github.com/spdx/spdx-spec/blo b/v2.2/chapters/appendix-II-license-matching-guidelines-and-templates.md
+String normalize(String text) {
+  // Guideline 4.1.1: upper case and lower case letters should be treated as lower case letters
+  text = text.toLowerCase();
+
+  text = text.replaceAll(_newLineRegex, '\n');
+
+  // Guideline 3.1.1: All whitespace should be treated as a single blank space.
+  text = text.replaceAll(_horizontalWhiteSpaceRegex, ' ');
+
+  // Guideline 5: Equivalent Punctuation marks
+  _equivalentPunctuationMarks.forEach((reg, value) {
+    text = text.replaceAll(reg, value);
+  });
+
+  // Guideline 8.1.1: Legally equal words must be treated same.
+  _varietalWords.forEach((key, value) {
+    text = text.replaceAll(RegExp(key), value);
+  });
+
+  // Guideline 2.1.4: Omitable texts that can be ignored.
+  text = _removeOmmitableText(text);
+  return text;
+}
+
+final _horizontalWhiteSpaceRegex = RegExp(r'[^\S\r\n]+');
+final _newLineRegex = RegExp('[\r\u0085]');
```

Comment:
is `\u0085` different from `\n` ?

Reply:
Yes, they are different \u0085 is `NEL`. I don't think it is used nowadays but I just added it as a precaution.